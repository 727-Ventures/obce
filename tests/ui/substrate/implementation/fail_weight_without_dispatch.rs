mod test_pallet;

pub struct ChainExtension;

#[obce::definition]
pub trait ChainExtensionDefinition {
    fn extension_method(&mut self);
}

#[obce::implementation]
impl<'a, E, T, Env> ChainExtensionDefinition for ExtensionContext<'a, E, T, Env, ChainExtension>
where
    T: SysConfig + ContractConfig + crate::test_pallet::Config,
    <<T as SysConfig>::Lookup as StaticLookup>::Source: From<<T as SysConfig>::AccountId>,
    Env: ChainExtensionEnvironment<E, T>,
    E: Ext<T = T>
{
    #[obce(weight())]
    fn extension_method(&mut self) {
        crate::test_pallet::Pallet::<T>::test_method(
            RawOrigin::Signed(self.env.ext().address().clone()).into(),
            123
        ).unwrap();
    }
}

fn main() {}
