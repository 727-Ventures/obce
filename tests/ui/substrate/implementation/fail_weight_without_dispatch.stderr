error: either "dispatch" or "expr" attributes are expected
  --> tests/ui/substrate/implementation/fail_weight_without_dispatch.rs:19:12
   |
19 |     #[obce(weight())]
   |            ^^^^^^^^
